src/types.ts

```diff
@@ -0,0 +1,404 @@
+            case ProductActionType.RemoveFromCart:
+                return 'Remove from Cart';
+            case ProductActionType.Checkout:
+                return 'Checkout';
+            case ProductActionType.CheckoutOption:
+                return 'Checkout Option';
+            case ProductActionType.Click:
+                return 'Click';
+            case ProductActionType.ViewDetail:
+                return 'View Detail';
+            case ProductActionType.Purchase:
+                return 'Purchase';
+            case ProductActionType.Refund:
+                return 'Refund';
+            case ProductActionType.AddToWishlist:
+                return 'Add to Wishlist';
+            case ProductActionType.RemoveFromWishlist:
+                return 'Remove from Wishlist';
+            default:
+                return 'Unknown';
+        }
+    },
+
+    // these are the action names used by server and mobile SDKs when expanding a CommerceEvent
+    getExpansionName: (id: number) => {
+        switch (id) {
+            case ProductActionType.AddToCart:
+                return 'add_to_cart';
+            case ProductActionType.RemoveFromCart:
+                return 'remove_from_cart';
+            case ProductActionType.Checkout:
+                return 'checkout';
+            case ProductActionType.CheckoutOption:
+                return 'checkout_option';
+            case ProductActionType.Click:
+                return 'click';
+            case ProductActionType.ViewDetail:
+                return 'view_detail';
+            case ProductActionType.Purchase:
+                return 'purchase';
+            case ProductActionType.Refund:
+                return 'refund';
+            case ProductActionType.AddToWishlist:
+                return 'add_to_wishlist';
+            case ProductActionType.RemoveFromWishlist:
+                return 'remove_from_wishlist';
+            default:
+                return 'unknown';
+        }
+    },
+};
+
+export const PromotionActionType = {
+    Unknown: 0 as const,
+    PromotionView: 1 as const,
+    PromotionClick: 2 as const,
+
+    getName: (id: number): string => {
+        switch (id) {
+            case PromotionActionType.PromotionView:
+                return 'view';
+            case PromotionActionType.PromotionClick:
+                return 'click';
+            default:
+                return 'unknown';
+        }
+    },
+
+    // these are the names that the server and mobile SDKs use while expanding CommerceEvent
+    getExpansionName: (id: number): string => {
+        switch (id) {
+            case PromotionActionType.PromotionView:
+                return 'view';
+            case PromotionActionType.PromotionClick:
+                return 'click';
+            default:
+                return 'unknown';
+        }
+    },
+};
+
+export const ProfileMessageType = {
+    Logout: 3 as const,
+};
+
+export const ApplicationTransitionType = {
+    AppInit: 1 as const,
+};
```

Comment:
this could be moved to serverModel.ts since it's only called there.  and that's already a .ts file unlike the identity.js comment above

Reply:
Same as above. I think this should stay in here since it's a data type.